larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -239,6 +239,42 @@ const MCParticle *LArMCParticleHelper::GetParentMCParticle(const MCParticle *con
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+void LArMCParticleHelper::GetAllDescendentMCParticles(const pandora::MCParticle *const pMCParticle,
+        pandora::MCParticleList& descendentMCParticleList)
+{
+    MCParticleList daughterMCParticleList = pMCParticle->GetDaughterList();
+    for (const MCParticle* pDaughterMCParticle : daughterMCParticleList)
+    {
+        if (std::find(descendentMCParticleList.begin(), descendentMCParticleList.end(), pDaughterMCParticle) ==
+                descendentMCParticleList.end())
+        {
+            descendentMCParticleList.push_back(pDaughterMCParticle);
+            LArMCParticleHelper::GetAllDescendentMCParticles(pDaughterMCParticle, descendentMCParticleList);
+        }
+    }
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void LArMCParticleHelper::GetAllAncestorMCParticles(const pandora::MCParticle *const pMCParticle,
+        pandora::MCParticleList& ancestorMCParticleList)
```

Comment:
Tend to prefer reference `&` to be next to variable, rather than the type. Check indentation.